webapi/Services/MemoryMigration/ChatMigrationStatus.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace CopilotChat.WebApi.Services.MemoryMigration;
+
+/// <summary>
+/// Set of migration states/status for chat memory migration.
+/// </summary>
+/// <remarks>
+/// Interlocked.CompareExchange doesn't work with enums.
+/// </remarks>
+public sealed class ChatMigrationStatus
```

Comment:
nit: Would a enum work>

Reply:
That was my starting point.  Apparently, Interlocked.CompareExchange doesn't work with enums.